client/signup/steps/domains/test/utils.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * @jest-environment jsdom
+ */
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	getExternalBackUrl,
+	backUrlExternalSourceStepsOverrides,
+	backUrlSourceOverrides,
+} from '../utils';
+
+const backUrlSourceOverrideKeys = Object.keys( backUrlSourceOverrides );
+const backUrlSourceOverrideKey = backUrlSourceOverrideKeys[ 0 ];
+
+describe( 'getExternalBackUrl', () => {
+	test( 'it should return false if given no first source arg', () => {
+		const result = getExternalBackUrl( undefined );
+		expect( result ).toBe( false );
+		expect( result ).toBeFalsy();
+	} );
+
+	test( 'it should return false if given an unexpected first argument that is also an invalid url', () => {
+		const result = getExternalBackUrl( 'unexpected-string' );
+		expect( result ).toBe( false );
+		expect( result ).toBeFalsy();
+	} );
+
+	test( 'it should return correct value if the given source is a key on the overrides object', () => {
+		const result = getExternalBackUrl( backUrlSourceOverrideKey );
+		expect( result ).toBe( backUrlSourceOverrides[ backUrlSourceOverrideKey ] );
+		expect( result ).toBeTruthy();
```

Comment:
Basically the same question here. Why do we need to check for `truthy` as well?

Reply:
In the codebase the `result` is [coerced into a truthy/falsy value](https://github.com/Automattic/wp-calypso/pull/48175/files#diff-dac1653ca8a4fe77505424b91353f493df971ad43841fce152c3572a13738a65R710) to perform an operation on, so I think here it makes sense even though we are checking the value, to also check that it's truthy.

Reply:
Thanks for your explanation. I think that it makes sense here too 👍